### PR TITLE
Cli list users

### PR DESCRIPTION
**Pull request by TimeEngineer**
It seems it is needed to retrieve the users id somewhere (for example for sharing workspace)

### CODE REVIEW

File: cli/src/list_users.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    device: Option<String>,
+    /// Skip revoked users
+    #[arg(short, long, default_value_t)]
+    skip_revoked: bool,
+    /// Result offset
+    #[arg(short, long)]
+    offset: Option<usize>,
+    /// Result limit
+    #[arg(short, long)]
+    limit: Option<usize>,
```
Comment (FirelightFlagboy) [inline cli/src/list_users.rs:26]: I'm not sure if it's useful to provide that to the user.

What was your plan with this ?
How the user would have used those options ?

**Review by FirelightFlagboy — COMMENTED** [review]
I don't see tests for this sub-command, is it something you have forgotten ?

File: cli/src/list_users.rs (hunk 1 above)
Comment (TimeEngineer) [reply cli/src/list_users.rs:26]: Idk, it was present with `human_find`

Comment (FirelightFlagboy) [reply cli/src/list_users.rs:26]: Then how the user will use those options ?

Comment (TimeEngineer) [reply cli/src/list_users.rs:26]: When he needs to paginate :man_shrugging:

Comment (TimeEngineer) [reply cli/src/list_users.rs:26]: Do you suggest to remove it ?

Comment (FirelightFlagboy) [reply cli/src/list_users.rs:26]: I think it would be best to do so

File: cli/src/tests.rs [hunk 2]
```diff
@@ -558,6 +558,43 @@ async fn create_workspace(tmp_path: TmpPath) {
...
+    Command::cargo_bin("parsec_cli")
+        .unwrap()
+        .args([
+            "create-workspace",
+            "--device",
+            &alice.slughash(),
+            "--name",
+            "new-workspace",
+        ])
+        .unwrap();
```
Comment (FirelightFlagboy) [inline cli/src/tests.rs:587]: Why creating a workspace _magically_ added 3 users ?

File: cli/src/tests.rs [hunk 3]
```diff
@@ -558,6 +558,43 @@ async fn create_workspace(tmp_path: TmpPath) {
...
+
+    Command::cargo_bin("parsec_cli")
+        .unwrap()
+        .args(["list-users", "--device", &alice.slughash()])
+        .assert()
+        .stdout(predicates::str::contains(format!(
+            "Found {GREEN}3{RESET} user(s)",
+        )));
```
Comment (FirelightFlagboy) [inline cli/src/tests.rs:595]: Could you check the user list ?

Comment (TimeEngineer) [reply cli/src/tests.rs:595]: I can't know the users id

File: cli/src/tests.rs (hunk 2 above)
Comment (TimeEngineer) [reply cli/src/tests.rs:587]: I have to investigate I guess

File: cli/src/tests.rs (hunk 3 above)
Comment (FirelightFlagboy) [reply cli/src/tests.rs:601]: Is it possible to only check the username (handle) ?

Comment (TimeEngineer) [reply cli/src/tests.rs:601]: I prefer to know that there are all users :thinking:

File: cli/src/tests.rs (hunk 2 above)
Comment (TimeEngineer) [reply cli/src/tests.rs]: I found that a `sync` was missing, and the `create_workspace` `sync` it

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**